nbgrader/apps/assignapp.py

```diff
@@ -12,103 +13,152 @@
     LockCells,
     ComputeChecksums,
     SaveCells,
-    CheckCellMetadata
+    CheckCellMetadata,
+    ClearOutput,
 )
 
 aliases = {}
 aliases.update(nbconvert_aliases)
 aliases.update({
-    'assignment': 'AssignmentExporter.assignment_id',
-    'db': 'AssignmentExporter.db_url'
 })
 
 flags = {}
 flags.update(nbconvert_flags)
 flags.update({
-    'save-cells': (
-        {'AssignApp': {'save_cells': True}},
-        "Save information about grade cells into the database."
+    'no-db': (
+        {'SaveCells': {'enabled': False}},
+        "Do not save information about grade cells into the database."
+    ),
+    'create': (
+        {'AssignApp': {'create_assignment': True}},
+        "Create the assignment at runtime if it does not exist."
     )
 })
 
 class AssignApp(BaseNbConvertApp):
 
     name = Unicode(u'nbgrader-assign')
-    description = Unicode(u'Prepare a student version of an assignment by removing solutions')
+    description = Unicode(u'Produce the version of an assignment to be released to students.')
 
     aliases = Dict(aliases)
     flags = Dict(flags)
 
     examples = Unicode(dedent(
         """
-        Running `nbgrader assign` on a file by itself will produce a student
-        version of that file in the same directory. In this case, it will produce
-        "Problem 1.nbconvert.ipynb":
-        
-        > nbgrader assign "Problem 1.ipynb"
+        Produce the version of the assignment that is intended to be released to
+        students. This performs several modifications to the original assignment:
 
-        If you want to override the .nbconvert part of the filename, you can use
-        the --output flag:
+            1. It inserts a header and/or footer to each notebook in the
+               assignment, if the header/footer are specified.
 
-        > nbgrader assign "Problem 1.ipynb" --output "Problem 1.student.ipynb"
+            2. It locks certain cells so that they cannot be deleted by students
+               accidentally (or on purpose!)
 
-        Or, you can put the student version in a different directory. In the
-        following example, there will be a file "student/Problem 1.ipynb" after
-        running `nbgrader assign`:
+            3. It removes solutions from the notebooks and replaces them with
+               code or text stubs saying (for example) "YOUR ANSWER HERE".
 
-        > nbgrader assign "Problem 1.ipynb" --build-dir=student
+            4. It clears all outputs from the cells of the notebooks.
 
-        You can also use shell globs, and copy files from one directory to another:
+            5. It computes checksums of the cells that contain tests for the
+               students' solutions, or that contain students' written answers.
+               This is done so that we can warn students if they have changed
+               the tests, or if they have failed to provide a response to a
+               written answer.
```

Comment:
I would flip the sentense around. Most user don't care if it computes checksum, which is an implementation detail. They want to know that changed cell will raise a warning.